src/EFCore.PG/Query/Expressions/Internal/PgFunctionExpression.cs

```diff
@@ -252,69 +273,84 @@ protected override Expression VisitChildren(ExpressionVisitor visitor)
                 : this;
         }
 
-        /// <summary>
-        ///     Tests if this object is considered equal to another.
-        /// </summary>
-        /// <param name="obj"> The object to compare with the current object. </param>
-        /// <returns>
-        ///     true if the objects are considered equal, false if they are not.
-        /// </returns>
-        public override bool Equals(object obj)
+        /// <inheritdoc />
+        public override bool Equals(object obj) => obj is PgFunctionExpression pgFunction && Equals(pgFunction);
+
+        /// <inheritdoc />
+        public bool Equals(PgFunctionExpression other)
+            => other != null &&
+               Instance == other.Instance &&
```

Comment:
`Instance`, `FunctionName` and `Schema` should be use value comparison (`Equals`) rather than reference comprison, as it was before this PR.